stock_valued_picking_report/models/stock_pack_operation.py

```diff
@@ -53,42 +54,43 @@ def _compute_sale_order_line_fields(self):
 
     @api.multi
     def sale_lines_values(self, sale_lines):
-        if len(sale_lines) <= 1:
-            price_unit = sale_lines.price_unit
-            discount = sale_lines.discount
-        else:
-            sum_qty = 0.0
-            sum_price = 0.0
-            sum_discount = 0.0
-            for sale_line in sale_lines:
-                sum_qty += sale_line.product_uom_qty
-                sum_price += sale_line.price_unit * sale_line.product_uom_qty
-                sum_discount += sale_line.discount * sale_line.product_uom_qty
-            price_unit = sum_price / (sum_qty or 1)
-            discount = sum_discount / (sum_qty or 1)
-        price_reduce = price_unit * (1 - (discount or 0.0) / 100.0)
+        sum_qty = 0.0
+        sum_price = 0.0
+        sum_discount = 0.0
+        sum_amount_untaxed = 0.0
+        sum_amount_tax = 0.0
+        sum_amount_total = 0.0
+        for sale_line in sale_lines:
+            sum_qty += sale_line.product_uom_qty
+            sum_price += sale_line.price_unit * sale_line.product_uom_qty
+            sum_discount += sale_line.discount * sale_line.product_uom_qty
+            amount_untaxed = sale_line.price_subtotal
+            sum_amount_untaxed += amount_untaxed
+            rm = sale_line.order_id.company_id.tax_calculation_rounding_method
```

Comment:
Do you need check this inside of the loop?

Reply:
The function take a set of sale order lines as an argument, I think this way let the function deal with sale order lines from different orders with different companies, but any suggestion is welcome :smile:

Reply:
Certainly it's not necessary so I put it outside to improve performance.